### PR TITLE
[SecurityBundle] Add json login ldap

### PR DESCRIPTION
**Pull request by ronfroy**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no   
| Deprecations? | no
| Tests pass?   | yes 
| License       | MIT

Add a simple from_login_ldap on firewall types to let authenticate with ldap with json API

### CODE REVIEW

**Comment by chalasr** [comment]
This needs some tests, at least a unit one for the factory, at best a functional one.

**Comment by chalasr** [comment]
> others LDAP SecurityListenerFactory are not tested

@ronfroy that's too bad but still, new code should be tested, at least ensuring that the container can compile with your addition. That should not be hard, adding a `json_login_ldap` listener to https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/SecurityBundle/Tests/DependencyInjection/Fixtures/yml/container1.yml (and xml + php) and updating the expected config in the corresponding tests that will fail should be enough.

**Review by pierre-tranchard — APPROVED** [review]

**Comment by ronfroy** [comment]
@chalasr done

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/JsonLoginLdap/config.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                default_roles: ROLE_USER
+                uid_key: uid
+
+    firewalls:
+        main:
+            pattern:  ^/login
+            stateless: true
+            anonymous: true
+            json_login_ldap:
+                check_path:               /login
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/JsonLoginLdap/config.yml]: I don't think we align values usually.

**Review by nicolas-grekas — COMMENTED** [review]
worth a line in the CHANGELOG file of the bundle - an also a doc PR? LGTM otherwise, thanks.

**Comment by ronfroy** [comment]
@nicolas-grekas done 
https://github.com/symfony/symfony-docs/pull/9987/files

**Comment by ronfroy** [comment]
@nicolas-grekas this PR https://github.com/symfony/symfony/pull/27695
Is more useful, to have a complete LDAP integration capability with user roles

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/JsonLogin/config.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ security:
             pattern: ^/
             anonymous: true
             json_login:
-               check_path:    /chk
+               check_path: /chk
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/JsonLogin/config.yml]: @ronfroy If this has to be changed, it must be done on the lowest branch where it exists to prevent merge conflicts (but I don't think it's worth it, we should just keep it in mind for new code). Can you please revert this change? Same for the `custom_handlers.yml` file below

**Comment by ronfroy** [comment]
@chalasr done

**Comment by chalasr** [comment]
Thank you @ronfroy.

**Comment by chalasr** [comment]
@ronfroy Note that github does not recognize your local git user so your contribs aren't linked to your github profile. See https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/ for fixing it
